pkg/operator/baremetal_config.go

```diff
@@ -0,0 +1,57 @@
+package operator
+
+import (
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	dynamic "k8s.io/client-go/dynamic"
+)
+
+var provisioningGVR = schema.GroupVersionResource{Group: "metal3.io", Resource: "provisionings", Version: "v1alpha1"}
+
+func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) BaremetalConfig {
+	provisioningClient := dc.Resource(provisioningGVR)
+	provisioningConfig, err := provisioningClient.Get(configName, metav1.GetOptions{})
+	if err != nil {
+		glog.V(1).Infof("Error getting provisioning config %q", configName)
```

Comment:
I find it weird to return `BaremetalConfig{}` here, at a minimum this should log error/warning rather than info

Reply:
Do you suggest a retry mechanism (good for transient errors) or a mechanism to bubble up an error (should not ignore errors) from this method?

Reply:
Done.